ecoli/analysis/multivariant/cell_mass.py

```diff
@@ -0,0 +1,162 @@
+1. each variant has its own plot;
+2. at each subplot, time is divided by generation id;
+
+It can also be used at multigeneration analysis.
+"""
+
+import os
+from typing import Any
+import altair as alt
+import polars as pl
+import pandas as pd
+from duckdb import DuckDBPyConnection
+
+
+def plot(
+    params: dict[str, Any],
+    conn: DuckDBPyConnection,
+    history_sql: str,
+    config_sql: str,
+    success_sql: str,
+    sim_data_dict: dict[str, dict[int, str]],
+    validation_data_paths: list[str],
+    outdir: str,
+    variant_metadata: dict[str, dict[int, Any]],
+    variant_names: dict[str, str],
+):
+    # Load data with required columns
+    required_columns = [
+        "time",
+        "variant",
+        "generation",
+        "agent_id",
+        "listeners__mass__dry_mass",
+    ]
+
+    sql = f"""
+    SELECT {", ".join(required_columns)}
+    FROM ({history_sql})
+    WHERE agent_id = 0
+    ORDER BY variant, generation, time
+    """
+
+    df = conn.sql(sql).pl()
+
+    # Process time and mass data
+    df = df.with_columns(
+        [
+            (pl.col("time") / 60).alias("time_min"),
+        ]
+    )
+
+    # Calculate initial mass for each generation for normalization
+    generation_stats = df.group_by(["variant", "generation"]).agg(
+        [pl.col("listeners__mass__dry_mass").min().alias("initial_mass")]
+    )
+
+    # Join back to main dataframe and calculate normalized mass
+    df = df.join(generation_stats, on=["variant", "generation"], how="left")
+
+    # Calculate normalized mass
+    df = df.with_columns(
+        [
+            (pl.col("listeners__mass__dry_mass") / pl.col("initial_mass")).alias(
+                "dry_mass_normalized"
+            )
+        ]
+    )
+
+    # Get variants and create plots
+    variants = df.select("variant").unique().to_series().to_list()
+
+    # ----------------------------------------#
+    plots = []
+
+    for variant in variants:
+        variant_df = df.filter(pl.col("variant") == variant).to_pandas()
+        variant_name = variant_names.get(variant, f"Variant {variant}")
+
+        # Create base chart with line plots only
+        base = alt.Chart(variant_df).add_selection(
+            alt.selection_interval(bind="scales")
+        )
+
+        # Base encoding
+        tooltip_fields: list[str] = ["time_min:Q", "generation:N"]
+        base_encode = {
+            "x": alt.X("time_min:Q", title="Time (min)", scale=alt.Scale(nice=False)),
+            "color": alt.Color(
+                "generation:N",
+                legend=alt.Legend(title="Generation"),
+                scale=alt.Scale(scheme="category10"),
+            ),
+        }
+
+        # Absolute dry mass plot
+        mass_plot = (
+            base.mark_line(strokeWidth=2.5)
+            .encode(
+                x=base_encode["x"],
+                color=base_encode["color"],
+                tooltip=tooltip_fields + ["listeners__mass__dry_mass:Q"],
```

Comment:
Separate out the lines for different seeds.
```suggestion
                tooltip=tooltip_fields + ["listeners__mass__dry_mass:Q"],
                detail="lineage_seed:N",
```